src/WebJobs.Extensions.DurableTask.Analyzers/SyntaxNodeUtils.cs

```diff
@@ -157,23 +158,48 @@ internal static bool TryGetFunctionNameAndNode(SemanticModel semanticModel, Synt
             return false;
         }
 
-        private static bool TryGetFunctionName(SemanticModel semanticModel, SyntaxNode attributeArgument, out string functionName)
+        public static bool TryGetFunctionName(SemanticModel semanticModel, SyntaxNode node, out string functionName)
         {
-            var stringLiteralExpression = attributeArgument.ChildNodes().Where(x => x.IsKind(SyntaxKind.StringLiteralExpression)).FirstOrDefault();
-            if (stringLiteralExpression != null)
+            if (TryGetStringFunctionName(node, out functionName))
             {
-                var stringLiteralToken = stringLiteralExpression.ChildTokens().Where(x => x.IsKind(SyntaxKind.StringLiteralToken)).FirstOrDefault();
-                if (stringLiteralToken != null)
+                return true;
+            }
+
+            if (TryGetNameOfFunctionName(node, out functionName))
```

Comment:
At initial glance, this name is very confusing. I got it eventually, and I get the naming convention you used to derive it, but I think we may need to stir this up a bit to make it a bit less confusing.

Consider `TryGetFunctionNameInNameOfOperator`. It's more verbose, but I think it's a bit more clear. It's up to you if you want to change the other method names to match (i.e. `TryGetFunctionNameInStringLiteral` and `TryGetFunctionNameInConstant`).

Reply:
I like this suggestion, thanks!